Source/objects.cpp

```diff
@@ -4349,6 +4350,60 @@ bool Object::IsDisabled() const
 	return IsAnyOf(static_cast<shrine_type>(_oVar1), shrine_type::ShrineFascinating, shrine_type::ShrineOrnate, shrine_type::ShrineSacred);
 }
 
+Object *ObjectAtPosition(Point position, bool considerLargeObjects)
+{
+	if (!InDungeonBounds(position)) {
+		return nullptr;
+	}
+
+	auto objectId = dObject[position.x][position.y];
+
+	if (objectId > 0 || (considerLargeObjects && objectId != 0)) {
+		return &Objects[abs(objectId) - 1];
+	}
+
+	// nothing at this position, return a nullptr
+	return nullptr;
+}
+
+bool IsObjectAtPosition(Point position)
+{
+	return ObjectAtPosition(position) != nullptr;
+}
+
+bool IsItemBlockingObjectAtPosition(Point position)
+{
+	Object *object = ObjectAtPosition(position);
+	if (object != nullptr && object->_oSolidFlag) {
+		// solid object
+		return true;
+	}
+
+	object = ObjectAtPosition(position + Direction::South);
+	if (object != nullptr && object->_oSelFlag != 0) {
+		// open single tile door
+		return true;
+	}
+
+	object = ObjectAtPosition(position + Direction::SouthEast, false);
+	if (object != nullptr) {
+		Object *otherDoor = ObjectAtPosition(position + Direction::SouthWest, false);
+		if (otherDoor != nullptr && object->_oSelFlag != 0 && otherDoor->_oSelFlag != 0) {
+			// open double tile door
+			return true;
+		}
+	}
+
+	return false;
+}
+
+bool IsMissileBlockingObjectAtPosition(Point position)

```

Comment:
```suggestion
bool IsMissilesBlockedByObjectAtPosition(Point position)
```

Reply:
This makes me think this function would be better defined in missiles.cpp. I'll update and move that definition.

Reply:
I ended up going with the name `IsMissileBlockedByTile`, renaming the existing `CouldMissileCollide` function. Every time a position was checked for an object with `missFlag` the code also checked `nMissileTable` so there was no point keeping them separate.